File: Leetcode/169/169.cpp
class Solution {
public:
    int majorityElement(vector<int> &nums) {
        unordered_map<int, int> A;
        for (const auto &i : nums) {
            ++A[i];
        }
        int len = static_cast<int>(nums.size());
        int res = nums.at(0);
        for (const auto it : A) {
            if (it.second > len / 2) {
                res = it.first;
                break;
            }
        }
        return res;
    }
};

File: Leetcode/169/169.go
func majorityElement(nums []int) int {
	val := math.MaxInt64
	counter := 0
	for i := 0; i < len(nums); i++ {
		if val != nums[i] {
			counter--
			if counter < 0 {
				val = nums[i]
				counter = 1
			}
		} else {
			counter++
		}
	}
	return val
}
